mono/mini/exceptions-amd64.c

```diff
@@ -885,260 +897,878 @@ mono_arch_exceptions_init (void)
 		  (the number in the OpInfo member) times 8 plus 8*/
+		unwindcode->UnwindOp = UWOP_ALLOC_SMALL;
 		unwindcode->OpInfo = (size - 8)/8;
-		unwindcode->UnwindOp = 2; /*UWOP_ALLOC_SMALL*/
 	}
 	else {
 		if (codesneeded == 3) {
 			/*the unscaled size of the allocation is recorded
-			  in the next two slots in little-endian format*/
-			*((unsigned int*)(&unwindcode->FrameOffset)) = size;
-			unwindcode += 2;
+			  in the next two slots in little-endian format.
+			  NOTE, unwind codes are allocated from end to begining of list so
+			  unwind code will have right execution order. List is sorted on CodeOffset
+			  using descending sort order.*/
+			unwindcode->UnwindOp = UWOP_ALLOC_LARGE;
 			unwindcode->OpInfo = 1;
+			*((unsigned int*)(&(unwindcode + 1)->FrameOffset)) = size;
 		}
 		else {
 			/*the size of the allocation divided by 8
-			  is recorded in the next slot*/
-			unwindcode->FrameOffset = size/8; 
-			unwindcode++;	
+			  is recorded in the next slot.
+			  NOTE, unwind codes are allocated from end to begining of list so
+			  unwind code will have right execution order. List is sorted on CodeOffset
+			  using descending sort order.*/
+			unwindcode->UnwindOp = UWOP_ALLOC_LARGE;
 			unwindcode->OpInfo = 0;
-			
+			(unwindcode + 1)->FrameOffset = (gushort)(size/8);
 		}
-		unwindcode->UnwindOp = 1; /*UWOP_ALLOC_LARGE*/
 	}
 
-	unwindcode->CodeOffset = (((guchar*)nextip)-((guchar*)codebegin));
-
-	if (unwindinfo->unwindInfo.SizeOfProlog >= unwindcode->CodeOffset)
+	if (unwindinfo->SizeOfProlog >= unwindcode->CodeOffset)
 		g_error ("Adding unwind info in wrong order.");
-	
-	unwindinfo->unwindInfo.SizeOfProlog = unwindcode->CodeOffset;
+
+	unwindinfo->SizeOfProlog = unwindcode->CodeOffset;
 }
 
-guint
-mono_arch_unwindinfo_get_size (gpointer monoui)
+static gboolean g_dyn_func_table_inited;
+
+// Dynamic function table used when registering unwind info for OS unwind support.
+static GList *g_dynamic_function_table_begin;
+static GList *g_dynamic_function_table_end;
+
+// SRW lock (lightweight read/writer lock) protecting dynamic function table.
+static SRWLOCK g_dynamic_function_table_lock = SRWLOCK_INIT;
+
+// Module handle used when explicit loading ntdll.
+static HMODULE g_ntdll;
+
+// If Win8 or Win2012Server or later, use growable function tables instead
+// of callbacks. Callback solution will still be fallback on older systems.
+static RtlAddGrowableFunctionTablePtr g_rtl_add_growable_function_table;
+static RtlGrowFunctionTablePtr g_rtl_grow_function_table;
+static RtlDeleteGrowableFunctionTablePtr g_rtl_delete_growable_function_table;
+
+// When using function table callback solution an out of proc module is needed by
+// debuggers in order to read unwind info from debug target.
+#ifdef _MSC_VER
+#define MONO_DAC_MODULE L"mono-2.0-dac-sgen.dll"
+#else
+#define MONO_DAC_MODULE L"mono-2.0-sgen.dll"
+#endif
+
+#define MONO_DAC_MODULE_MAX_PATH 1024
+
+static void
+mono_arch_unwindinfo_init_table_no_lock (void)
 {
```

Comment:
Used the same naming pattern for consistency, no problem to shorten the static function names. I will then drop mono_arch_unwindinfo prefix for these methods ending up with names like, init_table_no_lock, find_range_in_table_no_lock etc. OK?